.github/workflows/build_and_publish.yml

```diff
@@ -0,0 +1,76 @@
+name: Gatsby Publish
+
+on:
+  push:
+    branches:
+      [ main ]
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+  schedule: ## Do a run daily, to refresh website content
+    - cron: '0 3 * * *'
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  unit-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - run: npm install
+      - run: npm run test
+        env:
+          CI: true
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+      - run: npm install
+      - run: npm run build
+        env:
+          NODE_ENV: production
+          GATSBY_ACTIVE_ENV: production
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: npm run test:int
+        env:
+          CI: true
+      - name: Store PR id
+        run: echo ${{ github.event.number }} > ./public/pr-id.txt
+      - name: Publishing directory for PR preview
+        uses: actions/upload-artifact@v2
+        with:
+          name: site
+          path: ./public
+          retention-days: 3
+  deploy:
+    # Only try and deploy on merged code
+    if: "github.repository == 'quarkusio/extensions' && github.ref_name == 'main' && (github.event_name == 'push' || github.event_name == 'schedule')"
```

Comment:
```suggestion
    if: "github.repository == 'quarkiverse/quarkiverse' && github.ref_name == 'main' && (github.event_name == 'push' || github.event_name == 'schedule')"
```

Reply:
I wish we could use a variable for that, because I get it wrong every time I copy and paste. But of course the whole point is it has to be constant and not change when the repository name changes. It would be ideal if there was a variable like $is_fork or $upstream, but if there is, it's well hidden. 

I remembered to patch up the surge domains, but not that, thanks!